genesyscloud/outbound_contact_list_template/resource_genesyscloud_outbound_contact_list_template.go

```diff
@@ -0,0 +1,624 @@
+}
+
+func readOutboundContactListTemplate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	outboundApi := platformclientv2.NewOutboundApiWithConfig(sdkConfig)
+	cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceOutboundContactListTemplate(), constants.DefaultConsistencyChecks, resourceName)
+
+	log.Printf("Reading Outbound Contact List Template %s", d.Id())
+
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		sdkContactListTemplate, resp, getErr := outboundApi.GetOutboundContactlisttemplate(d.Id())
+		if getErr != nil {
+			if util.IsStatus404(resp) {
+				return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("failed to read Outbound Contact List Template %s | error: %s", d.Id(), getErr), resp))
+			}
+			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("failed to read Outbound Contact List Template %s | error: %s", d.Id(), getErr), resp))
+		}
+
+		if sdkContactListTemplate.Name != nil {
+			_ = d.Set("name", *sdkContactListTemplate.Name)
+		}
+		if sdkContactListTemplate.ColumnNames != nil {
+			var columnNames []string
+			for _, name := range *sdkContactListTemplate.ColumnNames {
+				columnNames = append(columnNames, name)
+			}
+			_ = d.Set("column_names", columnNames)
+		}
+		if sdkContactListTemplate.PhoneColumns != nil {
+			_ = d.Set("phone_columns", flattenSdkOutboundContactListTemplateContactPhoneNumberColumnSlice(*sdkContactListTemplate.PhoneColumns))
+		}
+		if sdkContactListTemplate.EmailColumns != nil {
+			_ = d.Set("email_columns", flattenSdkOutboundContactListTemplateContactEmailAddressColumnSlice(*sdkContactListTemplate.EmailColumns))
+		}
+		if sdkContactListTemplate.PreviewModeColumnName != nil {
+			_ = d.Set("preview_mode_column_name", *sdkContactListTemplate.PreviewModeColumnName)
+		}
+		if sdkContactListTemplate.PreviewModeAcceptedValues != nil {
+			var acceptedValues []string
+			for _, val := range *sdkContactListTemplate.PreviewModeAcceptedValues {
+				acceptedValues = append(acceptedValues, val)
+			}
+			_ = d.Set("preview_mode_accepted_values", acceptedValues)
+		}
+		if sdkContactListTemplate.AttemptLimits != nil && sdkContactListTemplate.AttemptLimits.Id != nil {
+			_ = d.Set("attempt_limit_id", *sdkContactListTemplate.AttemptLimits.Id)
+		}
+		if sdkContactListTemplate.AutomaticTimeZoneMapping != nil {
+			_ = d.Set("automatic_time_zone_mapping", *sdkContactListTemplate.AutomaticTimeZoneMapping)
+		}
+		if sdkContactListTemplate.ZipCodeColumnName != nil {
+			_ = d.Set("zip_code_column_name", *sdkContactListTemplate.ZipCodeColumnName)
+		}
+		if sdkContactListTemplate.ColumnDataTypeSpecifications != nil {
+			_ = d.Set("column_data_type_specifications", flattenSdkOutboundContactListTemplateColumnDataTypeSpecifications(*sdkContactListTemplate.ColumnDataTypeSpecifications))
+		}
```

Comment:
You could implement the generic functions from the `resourcedata` package for most of these